src/examples/create_datatype.cpp

```diff
@@ -0,0 +1,125 @@
+
+#include <iostream>
+
+#include <highfive/H5File.hpp>
+#include <highfive/H5DataType.hpp>
+
+using namespace HighFive;
+
+static const std::string FILE_NAME("create_datatype_example.h5");
+static const std::string DATASET_NAME("test_dataset");
+
+
+// Struct representation of custom type (type v below)
+typedef struct {
+    char a;
+    short b;
+    unsigned long long c;
+} csl;
+
+bool operator==(csl x, csl y) {
+    return x.a == y.a && x.b == y.b && x.c == y.c;
+}
+
+bool operator!=(csl x, csl y) {
+    return !(x == y);
+}
+
+// Tell HighFive how to create the HDF5 datatype for this base type by
+// specialising the create_datatype template
+namespace HighFive {
+template <> inline DataType create_datatype<csl>() {
+    CompoundType v_aligned({
+                               {"u1", AtomicType<unsigned char>{}},
+                               {"u2", AtomicType<short>{}},
+                               {"u3", AtomicType<unsigned long long>{}}
+    });
+    return std::move(v_aligned);
```

Comment:
move will avoid rvo

Reply:
gcc ask the move, because RVO seems impossible due to conversion `CompoundType` => `DataType`